packages/contracts-periphery/contracts/universal/faucet/Faucet.sol

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Semver } from "@eth-optimism/contracts-bedrock/contracts/universal/Semver.sol";
+import { SignatureChecker } from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import {
+    EIP712Upgradeable
+} from "@openzeppelin/contracts-upgradeable/utils/cryptography/draft-EIP712Upgradeable.sol";
+
+/**
+ * @title  SafeSend
+ * @notice Sends ETH to a recipient account without triggering any code.
+ */
+contract SafeSend {
+    /**
+     * @param recipient Account to send ETH to.
+     */
+    constructor(
+        address payable recipient
+    )
+        payable
+    {
+        selfdestruct(recipient);
+    }
+}
+
+/**
+ * @title  FaucetAuthModule
+ * @notice Interface for faucet authentication modules.
+ */
+interface FaucetAuthModule {
+    /**
+     * @notice Verifies that the given drip parameters are valid.
+     *
+     * @param params Drip parameters to verify.
+     * @param id     Authentication ID to verify.
+     * @param proof  Authentication proof to verify.
+     */
+    function verify(
+        Faucet.DripParameters memory params,
+        bytes memory id,
+        bytes memory proof
+    )
+        external
+        view
+        returns (
+            bool
+        );
+}
+
+/**
+ * @title  AdminFAM
+ * @notice FaucetAuthModule that allows an admin to sign off on a given faucet drip. Takes an admin
+ *         as the constructor argument.
+ */
+contract AdminFAM is FaucetAuthModule, Semver, EIP712Upgradeable {
```

Comment:
I'd break this out into its own file, same with the FaucetAuthModule interface, which I'm realizing should be `IFaucetAuthModule`